src/utils/time-utils.ts

```diff
@@ -150,30 +150,43 @@ const multipleVacationDaysSelected = (workingWeek: boolean[], vacationDayStart:
   const startsFromWorkingDay = workingWeek[vacationDayStart-1];
 
   if (vacationDayStart === vacationDayEnd) {
-    if (!startsFromWorkingDay) return weeks * 6;
-    // vacation starts and ends on working day
-    return weeks * 6 + 1;
+    return calculateVacationDurationInWeeks(startsFromWorkingDay, weeks);
   }
 
   if (vacationDayEnd > vacationDayStart) {
-    let takenWorkingDays = 0;
-    // calculate the number of vacation days from start till the end of vacation if the number of weeks is not an integer
-    for (let i = vacationDayStart; i <= vacationDayEnd; i++){
-      if (workingWeek[i-1]) takenWorkingDays+=1;
-    }
+    let takenWorkingDays = countWorkingWeekDaysInRange(vacationDayStart, vacationDayEnd, workingWeek);
     if (takenWorkingDays === workDays) takenWorkingDays = 6;
     return weeks * 6 + takenWorkingDays;
   }
 
+  let takenWorkingDays = countWorkingWeekDaysInRange(vacationDayStart, endWeek, workingWeek) + 
+  countWorkingWeekDaysInRange(startWeek, vacationDayEnd, workingWeek);
+  if (takenWorkingDays === workDays) takenWorkingDays = 6;
+  return weeks * 6 + takenWorkingDays;
+}
+
+/**
+ * Get vacation days based on the number of weeks, considering if it starts from a working day
+ *
+ * @param startsFromWorkingDay represents if choosen start date is the starting of work week
```

Comment:
Is this description correct or is it  "if the chosen start date is a working day"?